classes/Task/AbstractTask.php

```diff
@@ -200,8 +200,10 @@ public function init(): void
      */
     protected function setupEnvironment(): void
     {
-        if ($this::TASK_TYPE === TaskType::TASK_TYPE_UPDATE && $this->container->getUpgradeConfiguration()->isChannelLocal()) {
-            $archiveXml = $this->container->getUpgradeConfiguration()->getLocalChannelXml();
+        $updateConfiguration = $this->container->getConfigurationStorage()->loadUpdateConfiguration();
+
+        if ($this::TASK_TYPE === TaskType::TASK_TYPE_UPDATE && $updateConfiguration->isChannelLocal()) {
```

Comment:
```suggestion
        if ($this::TASK_TYPE === TaskType::TASK_TYPE_UPDATE && $this->container->getConfigurationStorage()->loadUpdateConfiguration()->isChannelLocal()) {
```

With this change, the update configure will be loaded ONLY if `$this::TASK_TYPE === TaskType::TASK_TYPE_UPDATE` is true, making the execution of this function a bit faster in the restore & backup process when called.